cdap-app-fabric/src/main/java/io/cdap/cdap/internal/events/SparkProgramStatusMetricsProvider.java

```diff
@@ -178,4 +180,28 @@ private String generateMaxTerminationDateParam() {
       .withZone(ZoneId.systemDefault());
     return targetDate.format(formatter);
   }
+
+  private void validateConfiguration() throws MetricRetrievalException {
+    String host = cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_HOST);
+    String typeKey = cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.TYPE);
+    String maxRetriesKey =
+      cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.MAX_RETRIES);
+    String maxTimeKey =
+      cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.MAX_TIME_SECS);
+    String baseDelayKey =
+      cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.DELAY_BASE_MS);
+    String maxDelayKey =
+      cConf.get(Constants.Spark.SPARK_METRICS_PROVIDER_RETRY_STRATEGY_PREFIX + Constants.Retry.DELAY_MAX_MS);
+
+    if (StringUtils.isNullOrEmpty(host)) {
+      throw new MetricRetrievalException("Failed to retrieves metrics for Spark program due to Spark metrics host is " +
+                                           "not set");
+    }
+    if (StringUtils.isNullOrEmpty(typeKey) || StringUtils.isNullOrEmpty(maxRetriesKey)
+      || StringUtils.isNullOrEmpty(maxTimeKey) || StringUtils.isNullOrEmpty(baseDelayKey)
+      || StringUtils.isNullOrEmpty(maxDelayKey)) {
+      throw new MetricRetrievalException("Failed to retrieves metrics for Spark program due to Spark metrics retry " +
+                                           "strategy is not properly set");
```

Comment:
Could we split this check and return a more explicit error message? (Listing what is not set, or why the strategy was not correctly set.

Reply:
Done